csharp/Adapter/Microsoft.Spark.CSharp/Services/LoggerServiceFactory.cs

```diff
@@ -31,7 +30,21 @@ public static void SetLoggerService(ILoggerService loggerServiceOverride)
         /// <returns>An instance of logger service</returns>
         public static ILoggerService GetLogger(Type type)
         {
-            return  loggerService.GetLoggerInstance(type);
+            return loggerService.Value.GetLoggerInstance(type);
+        }
+
+        /// <summary>
+        /// if there exists exe.config file, then use log4net
+        /// </summary>
+        /// <returns></returns>
+        private static ILoggerService GetDefaultLogger()
+        {
+            if (File.Exists(AppDomain.CurrentDomain.SetupInformation.ConfigurationFile))
```

Comment:
I think we can add a check for log4net section in the config file more specifically, such as (pseudo-code) - ConfigurationManager.GetSection("log4net") != null

https://msdn.microsoft.com/en-us/library/system.configuration.configurationmanager.getsection(v=vs.110).aspx